src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -1630,4 +1686,17 @@ private void logAttackMoves(final List<ProTerritory> prioritizedTerritories) {
       }
     }
   }
+
+  private boolean canAirSafelyLandAfterAttack(final Unit unit, final Territory t) {
+    final boolean isAdjacentToAlliedFactory =
+        Matches.territoryHasNeighborMatching(data,
+            ProMatches.territoryHasInfraFactoryAndIsAlliedLand(player, data)).match(t);
+    final int range = TripleAUnit.get(unit).getMovementLeft();
+    final int distance =
+        data.getMap().getDistance_IgnoreEndForCondition(ProData.unitTerritoryMap.get(unit), t,
+            ProMatches.territoryCanMoveAirUnitsAndNoAA(player, data, true));
+    final boolean usesMoreThanHalfOfRange = distance > range / 2;
+    return isAdjacentToAlliedFactory || !usesMoreThanHalfOfRange;
```

Comment:
Given this method is pretty simple and only a few lines and those helper methods would only be called from this method I don't think its worth breaking it up more. You could add an if to check them in series rather than parallel but I think that actually makes it harder to read/maintain.